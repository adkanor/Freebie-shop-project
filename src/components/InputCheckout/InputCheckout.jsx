import React from "react";
import { Field } from "formik";
import stylesForm from "../../pages/CheckOut/CheckOut.module.css";
import PropTypes from "prop-types";
import InputMask from "react-input-mask";

const InputCheckout = ({ name, text, type, isError, errorText, children }) => {
    const token = localStorage.getItem("token");

    const isEmailField = name === "email";
    const inputStyle = isEmailField
        ? { color: "var( --gray-text-primary)" }
        : {};
    return (
        <div className={stylesForm.formItem}>
            <label className={stylesForm.formLabel} htmlFor={name}>
                {text}
            </label>
            {type !== "phone" ? (
                <>
                    <Field
                        type={type}
                        id={name}
                        name={name}
                        className={stylesForm.formInput}
                        readOnly={token ? isEmailField : false}
                        style={inputStyle}
                    />
                    {children}
                </>
            ) : (
                <Field name={name}>
                    {({ field }) => (
                        <InputMask
                            {...field}
                            className={stylesForm.formInput}
                            mask="+38 (999) 999-99-99"
                        />
                    )}
                </Field>
            )}

            {isError ? (
                <p className={stylesForm.errorMessage}>{`*${errorText}`}</p>
            ) : null}
        </div>
    );
};

InputCheckout.propTypes = {
    children: PropTypes.node,
    name: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    isError: PropTypes.bool,
    errorText: PropTypes.string,
    type: PropTypes.string,
};

export default InputCheckout;
