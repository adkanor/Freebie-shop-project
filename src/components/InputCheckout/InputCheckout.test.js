import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Formik } from 'formik';
import InputCheckout from './InputCheckout';

describe('InputCheckout Component', () => {
    it('should render a label and an input field', () => {
        render(
          <Formik initialValues={{ testName: '' }} onSubmit={() => {}}>
            <InputCheckout name="testName" text="Test Label" />
          </Formik>
        );
        const label = screen.getByText('Test Label');
        const input = screen.getByLabelText('Test Label');
        expect(label).toBeInTheDocument();
        expect(input).toBeInTheDocument();
      });

  it('should display an error message when isError is true', () => {
    render(
      <Formik initialValues={{ testName: '' }} onSubmit={() => {}}>
        <InputCheckout name="testName" text="Test Label" isError={true} errorText="This is an error" />
      </Formik>
    );
    const errorMessage = screen.getByText('*This is an error');
    expect(errorMessage).toBeInTheDocument();
  });

  it('should not display an error message when isError is false', () => {
    render(
      <Formik initialValues={{ testName: '' }} onSubmit={() => {}}>
        <InputCheckout name="testName" text="Test Label" isError={false} errorText="This is an error" />
      </Formik>
    );
    const errorMessage = screen.queryByText('*This is an error');
    expect(errorMessage).toBeNull();
  });


  it('should render an input field with a phone mask when type is "phone"', () => {
    render(
      <Formik initialValues={{ testName: '' }} onSubmit={() => {}}>
        <InputCheckout name="testName" text="Test Label" type="phone" />
      </Formik>
    );

    const input = screen.getByRole('textbox');
    expect(input).toHaveValue('');
  });

  it('should match snapshot', () => {
    const { asFragment } = render(
      <Formik initialValues={{ testName: '' }} onSubmit={() => {}}>
        <InputCheckout name="testName" text="Test Label" />
      </Formik>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
