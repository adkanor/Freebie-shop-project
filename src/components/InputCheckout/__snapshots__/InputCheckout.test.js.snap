// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCheckout Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="formItem"
  >
    <label
      class="formLabel"
      for="testName"
    >
      Test Label
    </label>
    <input
      class="formInput"
      id="testName"
      name="testName"
      value=""
    />
  </div>
</DocumentFragment>
`;
